### PR TITLE
Updates to the quickstarts

### PR DESCRIPTION
**Pull request by josh146**
**Description of the Change:**

- Updates the quickstart reference guides to take into account the shots refactor.
- Adds the `qml.state` and `qml.dm` functions to the Measurements reference page
- Adds JAX to the homepage

**Benefits:** n/a

**Possible Drawbacks:** n/a

**Related GitHub Issues:** n/a

### CODE REVIEW

File: doc/introduction/interfaces/jax.rst [hunk 1]
```diff
@@ -3,15 +3,21 @@
 JAX interface
 =================
 
-Born out of the autograd package, `JAX <https://jax.readthedocs.io/en/latest/index.html>`_ is the next generation of differentiable functional computation, 
-adding support for powerful hardware accelerators like GPUs and TPUs via `XLA <https://www.tensorflow.org/xla>`_. In order to use PennyLane in combination with JAX, we have to generate JAX-compatible
-quantum nodes. A basic ``QNode`` can be translated into a quantum node that interfaces with JAX by using the ``interface='jax'`` flag in the QNode decorator.
+Born out of the autograd package, `JAX <https://jax.readthedocs.io/en/latest/index.html>`_ is the
+next generation of differentiable functional computation, adding support for powerful hardware
+accelerators like GPUs and TPUs via `XLA <https://www.tensorflow.org/xla>`_. In order to use
```
Comment (albi3ro) [inline doc/introduction/interfaces/jax.rst:8]:
```suggestion
accelerators like GPUs and TPUs via `XLA <https://www.tensorflow.org/xla>`_. To use
```
Unnecessary words

File: doc/introduction/measurements.rst [hunk 2]
```diff
@@ -162,14 +164,15 @@ it can sometimes be convenient to execute the same QNode with differing
 number of shots.
 
 For simulators like ``default.qubit``, finite shots will be simulated if
-we set ``analytic=False`` in the device.
+we set ``shots=None`` in the device.
```
Comment (albi3ro) [inline doc/introduction/measurements.rst:167]: Ummmm? I think this is a mistake here.  Did you mean something like:

```suggestion
we set ``shots`` to a positive integer.
```

File: doc/introduction/measurements.rst [hunk 3]
```diff
@@ -162,14 +164,15 @@ it can sometimes be convenient to execute the same QNode with differing
...
 
-The shot number can be changed by modifying the value of :attr:`.Device.shots`:
+The shot number can be changed on the device itself, or temporarily
+by specifying the ``shots`` keyword argument when executing the QNode:
```
Comment (albi3ro) [inline doc/introduction/measurements.rst:170]:
```suggestion
The shot number can be changed on the device itself, or temporarily altered
by the ``shots`` keyword argument when executing the QNode:
```
I think that sounds slightly better.

File: doc/introduction/measurements.rst [hunk 4]
```diff
@@ -201,23 +203,20 @@ circuit:
         qml.Hadamard(wires=0)
         return qml.expval(qml.PauliZ(0))
 
-Running the simulator when ``analytic`` is set to ``True`` returns the exact expectation.
+Running the simulator when ``shots=None`` returns the exact expectation.
```
Comment (albi3ro) [inline doc/introduction/measurements.rst:206]:
```suggestion
Running the simulator with ``shots=None`` returns the exact expectation.
```

**Review by albi3ro — CHANGES_REQUESTED** [review]
I gave several alternate wordings that might flow a bit smoother, but those comments aren't particularly important.

I will request that Line 167 of `measurements.rst` gets fixed, but everything else is optional.

File: doc/introduction/measurements.rst (hunk 2 above)
Comment (josh146) [reply doc/introduction/measurements.rst:167]: Good catch!!

**Comment by josh146** [comment]
Thanks @albi3ro -- all your suggestions were worthwhile, so I merged them all in!

File: doc/introduction/circuits.rst [hunk 5]
```diff
@@ -90,7 +92,7 @@ instantiated using the :func:`device <pennylane.device>` loader.
 
 .. code-block:: python
 
-    dev = qml.device('default.qubit', wires=2, shots=1000, analytic=False)
```
Comment (mariaschuld) [inline doc/introduction/circuits.rst:93]: Oh sorry, that must have slipped my search in the refactor!

File: doc/introduction/circuits.rst [hunk 6]
```diff
@@ -110,13 +112,11 @@ devices, the options are:
...
-
-* ``shots`` (*int*): How many times the circuit should be evaluated (or sampled) to estimate
-  the expectation values. Defaults to 1000 if not specified. This argument can be temporarily overwritten
-  when a QNode is called; for example ``my_qnode(shots=3)`` will temporarily evaluate ``my_qnode``
-  using three shots.
+* ``shots`` (*None* or *int* or *list[int]*): How many times the circuit should be evaluated (or
+  sampled) to estimate statistical quantities. On some supported simulator devices,
+  ``shots=None`` indicates to compute measurement statistics *exactly*. Note that this
+  argument can be temporarily overwritten when a QNode is called. For example, ``my_qnode(shots=3)``
+  will temporarily evaluate ``my_qnode`` using three shots.
```
Comment (mariaschuld) [inline doc/introduction/circuits.rst:119]: Nice!

File: doc/introduction/interfaces/jax.rst [hunk 7]
```diff
@@ -89,14 +95,14 @@ DeviceArray(-3.4332792e-10, dtype=float32)
 Using jax.jit on QNodes
 -----------------------
 
-To fully utilize the power and speed of JAX, you'll need to just-in-time compile
-your functions - a process called "jitting". 
-If you're only taking expectation values, you can simply add the decorator on your QNode directly.
+To fully utilize the power and speed of JAX, you'll need to just-in-time compile your functions - a
+process called "jitting". If you're only taking expectation values, you can simply add the decorator
+on your QNode directly.
```
Comment (mariaschuld) [inline doc/introduction/interfaces/jax.rst:100]: I thought you don't like the "you" in docs? :)

**Review by mariaschuld — APPROVED** [review]
Thanks for cleaning that up, and sorry for not finding these instances in the refactor!

File: doc/introduction/interfaces/jax.rst (hunk 7 above)
Comment (josh146) [reply doc/introduction/interfaces/jax.rst:100]: 🤔 Let me rephrase this.

Comment (josh146) [inline doc/introduction/interfaces/jax.rst:100]:
```suggestion
To fully utilize the power and speed of JAX, you'll need to just-in-time compile your functions - a
process called "jitting". If only expectation values are returned, the `@jax.jit` decorator can be
directly applied to the QNode.
```

File: doc/introduction/interfaces/jax.rst [hunk 8]
```diff
@@ -89,14 +95,14 @@ DeviceArray(-3.4332792e-10, dtype=float32)
 Using jax.jit on QNodes
 -----------------------
 
-To fully utilize the power and speed of JAX, you'll need to just-in-time compile
-your functions - a process called "jitting". 
-If you're only taking expectation values, you can simply add the decorator on your QNode directly.
+To fully utilize the power and speed of JAX, you'll need to just-in-time compile your functions - a
+process called "jitting". If only expectation values are returned, the `@jax.jit` decorator can be
```
Comment (josh146) [inline doc/introduction/interfaces/jax.rst:99]:
```suggestion
process called "jitting". If only expectation values are returned, the ``@jax.jit`` decorator can be
```
